slick/src/main/scala/slick/jdbc/SQLServerProfile.scala

```diff
@@ -107,8 +107,13 @@ trait SQLServerProfile extends JdbcProfile {
   override def createModelBuilder(tables: Seq[MTable], ignoreInvalidDefaults: Boolean)(implicit ec: ExecutionContext): JdbcModelBuilder =
     new ModelBuilder(tables, ignoreInvalidDefaults)
 
-  override def defaultTables(implicit ec: ExecutionContext): DBIO[Seq[MTable]] =
-    MTable.getTables(None, None, None, Some(Seq("TABLE")))
+  override def defaultTables(implicit ec: ExecutionContext): DBIO[Seq[MTable]] = {
+    import api._
+    for {
+      schema <- Functions.user.result
+      mtables <- MTable.getTables(None, Some(schema), None, Some(Seq("TABLE")))
```

Comment:
For SqlServer, schemas and users aren’t necessarily the same thing, but after SqlServer2008, there are system tables that come back from a query for tables that doesn’t filter by schema. The defaultTables method seems to be intended for list all the tables associated with the current connection and so when system tables come back, tests start to fail.